mcs/class/System.Configuration/Test/System.Configuration/ConfigurationManagerTest.cs

```diff
@@ -627,5 +627,26 @@ public void TestConnectionStringRetrieval ()
 			Assert.AreEqual ("Server=(local);Initial Catalog=someDb;User Id=someUser;Password=somePassword;Application Name=someAppName;Min Pool Size=5;Max Pool Size=500;Connect Timeout=10;Connection Lifetime=29;",
 			                 connString);
 		}
+
+		[Test]
+		public void BadConfig ()
+		{
+			string xml = @" badXml";
+
+			var file = Path.Combine (tempFolder, "badConfig.config");
+			File.WriteAllText (file, xml);
+
+			var failed = false;
+
+			try {
+				var fileMap = new ConfigurationFileMap (file);
+				var configuration = ConfigurationManager.OpenMappedMachineConfiguration (fileMap);
+			} catch (ConfigurationErrorsException e) {
+				failed = true;
+				Assert.AreEqual (file, e.Filename);
+			}
+
+			Assert.AreEqual (true, failed, "Exception ConfigurationErrorsException was expected.");
```

Comment:
Usual pattern is try block where last line is Assert.Fail ("Exception ConfigurationErrorsException was expected.")